### ISSUE TITLE
[Task] Port the transformer blocks from Transformers4Rec (PyTorch) to Merlin Models (Tensorflow)

### ISSUE THREAD

**Issue by karlhigley** [NVIDIA-Merlin/Merlin#276]
## Problem:

The stable API in Transformers4Rec is based on PyTorch and includes all components used to run session-based research experiments reported in T4Rec paper. On the other hand the Merlin Models does not support a stable PyTorch API yet. 

## Goal:

- The goal of this work is to port all t4rec blocks needed for defining transformer-based recommendation models (PyTorch implementation) into Merlin Models (Tensorflow implementation). 

- This work is not about improving t4rec current API or adding new blocks other than the existing ones 

## Constraints:

-  The Pytorch T4Rec API is inheriting from HuggingFace trainer class for supporting optimized techniques such as fp16, multi-gpu, early-stopping… We need to provide clear guidance of how to set up these techniques using the  Keras fit method. 

-  Current T4Rec implementation is using the schema class from old merlin_standardlib, the migration to use merlin-core should happen before porting to Merlin Models to makes sure all blocks are correctly working with the new Schema class.  

## Starting Point:

- [x] Implement MaskingBlock : Causal LM, Masked LM (https://github.com/NVIDIA-Merlin/models/pull/780)

- [x] Port the Transformer-block class related to HugginFace architectures adapted for next item prediction  (Link to HF layers)

- [x] Implementation of Transformer block  based in the configs defined in [HuggingFace - transformers](https://github.com/huggingface/transformers)-based library.
**Note:**  [This](https://github.com/huggingface/transformers/blob/fde22c75a157a9f8c2da620a3b3bf9a23b5b3a0a/src/transformers/models/bert/modeling_tf_bert.py#L437) is an example of how Transformer-based architectures are implemented in HF (as keras layers). 

- Support setting the masking task within model.compile() => Not needed in the new API design. 

22.11 release: 

- [ ] NVIDIA-Merlin/models#791

**Comment by viswa-nvidia** [comment]
@sararb , could you please help to define this ticket.

**Comment by sararb** [comment]
This issue is fixed in [#771](https://github.com/NVIDIA-Merlin/models/pull/771) and [#780](https://github.com/NVIDIA-Merlin/models/pull/780)

**Comment by sararb** [comment]
I moved the task of creating the example to a separate ticket NVIDIA-Merlin/models#791 (expected in 22.11)
